aws/resource_aws_batch_compute_environment.go

```diff
@@ -0,0 +1,428 @@
+		if len(computeResources) == 0 {
+			return fmt.Errorf("One compute environment is expected, but no compute environments are set")
+		}
+		computeResource := computeResources[0].(map[string]interface{})
+
+		instanceRole := computeResource["instance_role"].(string)
+		maxvCpus := int64(computeResource["max_vcpus"].(int))
+		minvCpus := int64(computeResource["min_vcpus"].(int))
+		computeResourceType := computeResource["type"].(string)
+
+		var instanceTypes []*string
+		for _, v := range computeResource["instance_type"].(*schema.Set).List() {
+			instanceTypes = append(instanceTypes, aws.String(v.(string)))
+		}
+
+		var securityGroupIds []*string
+		for _, v := range computeResource["security_group_ids"].(*schema.Set).List() {
+			securityGroupIds = append(securityGroupIds, aws.String(v.(string)))
+		}
+
+		var subnets []*string
+		for _, v := range computeResource["subnets"].(*schema.Set).List() {
+			subnets = append(subnets, aws.String(v.(string)))
+		}
+
+		input.ComputeResources = &batch.ComputeResource{
+			InstanceRole:     aws.String(instanceRole),
+			InstanceTypes:    instanceTypes,
+			MaxvCpus:         aws.Int64(maxvCpus),
+			MinvCpus:         aws.Int64(minvCpus),
+			SecurityGroupIds: securityGroupIds,
+			Subnets:          subnets,
+			Type:             aws.String(computeResourceType),
+		}
+
+		if v, ok := computeResource["bid_percentage"]; ok {
+			input.ComputeResources.BidPercentage = aws.Int64(int64(v.(int)))
+		}
+		if v, ok := computeResource["desired_vcpus"]; ok {
+			input.ComputeResources.DesiredvCpus = aws.Int64(int64(v.(int)))
+		}
+		if v, ok := computeResource["ec2_key_pair"]; ok {
+			input.ComputeResources.Ec2KeyPair = aws.String(v.(string))
+		}
+		if v, ok := computeResource["image_id"]; ok {
+			input.ComputeResources.ImageId = aws.String(v.(string))
+		}
+		if v, ok := computeResource["spot_iam_fleet_role"]; ok {
+			input.ComputeResources.SpotIamFleetRole = aws.String(v.(string))
+		}
+		if v, ok := computeResource["tags"]; ok {
+			input.ComputeResources.Tags = tagsFromMapGeneric(v.(map[string]interface{}))
+		}
+	}
+
+	if _, err := conn.CreateComputeEnvironment(input); err != nil {
+		return err
+	}
+
+	d.SetId(computeEnvironmentName)
+
+	if err := waitComputeEnvironmentValid(d, meta); err != nil {
+		return err
+	}
+
+	return resourceAwsBatchComputeEnvironmentRead(d, meta)
+}
+
+func resourceAwsBatchComputeEnvironmentRead(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).batchconn
+
+	computeEnvironmentName := d.Get("compute_environment_name").(string)
+
+	input := &batch.DescribeComputeEnvironmentsInput{
+		ComputeEnvironments: []*string{
+			aws.String(computeEnvironmentName),
+		},
+	}
+
+	result, err := conn.DescribeComputeEnvironments(input)
+	if err != nil {
+		return err
+	}
+
+	if len(result.ComputeEnvironments) == 0 {
+		return fmt.Errorf("One compute environment is expected, but AWS return no compute environment")
+	} else if len(result.ComputeEnvironments) >= 2 {
```

Comment:
This should be on a newline, since the return instruction line 245 returns
Also, is it possible to have 2 ComputeEnvironments with the same name?